src/main/java/org/nastya/service/UserService.java

```diff
@@ -36,7 +37,7 @@ public boolean existsByUsername(String username) {
         return exists;
     }
 
-    public UserDTO saveUser(UserDTO userDTO) throws UserAlreadyExistsException {
+    public synchronized UserDTO saveUser(UserDTO userDTO, HttpHeaders headers) throws UserAlreadyExistsException {
```

Comment:
Did you forget what I told you about synchronized and reentrantlock? It is not good enough

Reply:
I have already wrote and told you how it has to be fixed.